Meter.h

```diff
@@ -97,7 +96,8 @@ typedef struct MeterClass_ {
 
 typedef struct GraphData_ {
    struct timeval time;
-   double values[METER_GRAPHDATA_SIZE];
+   unsigned int nValues;
```

Comment:
Prefer `size_t` over `unsigned int`, that way the type also conveys some meaning of the contained value.

Reply:
IMO that would convey the _wrong_ meaning.  `size_t` is for object sizes (i.e. a number of bytes), but that's not what this is -- it's the number of elements in the `values` array, which I think is reasonably clear as it stands.

Reply:
Let me quote [cpp-reference.com](https://en.cppreference.com/w/c/types/size_t):
> Notes
> `size_t` can store the maximum size of a theoretically possible object of any type (including array).
>
> **`size_t` is commonly used for array indexing and loop counting.** Programs that use other types, such as `unsigned int`, for array indexing may fail on, e.g. 64-bit systems when the index exceeds [`UINT_MAX`](https://en.cppreference.com/w/c/types/limits) or if it relies on 32-bit modular arithmetic.

(Emphasis mine)

Reply:
Commonality notwithstanding I still find it actively misleading, but if you insist...